runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -814,7 +893,58 @@ void  chpl_comm_put(void* addr, c_nodeid_t node, void* raddr,
       chpl_comm_commDiagnostics.put++;
       chpl_sync_unlock(&chpl_comm_diagnostics_sync);
     }
-    gasnet_put(node, raddr, addr, size); // node, dest, src, size
+
+    // Handle remote address not in remote segment.
+#ifdef GASNET_SEGMENT_EVERYTHING
```

Comment:
These are two different questions and we should have two different functions to answer them. Although it seems like there are 3 cases, I believe under some circumstances there are actually 4. For segment=fast or =everything, where the address is in the GASNet segment but is in a page that doesn't physically exist yet because it hasn't ever been referenced, it would be in-segment but not referenceable, at least for a read. (You could write to it and this would cause it to come into existence, but reading from it would produce a segfault. Stack autogrow regions are like this.) So, I would be in favor of  renaming the current function as, say, chpl_comm_addr_exists(), and naming the new one chpl_comm_is_in_segment().

However, I don't think it's necessary or desirable to mix that change up with this one. You could either do a separate PR to rename the existing function first, or (much easier) do this PR first, as is, and then split the functions apart afterward.